rustc_tools_util/src/lib.rs

```diff
@@ -99,20 +102,36 @@ impl std::fmt::Debug for VersionInfo {
     }
 }
 
+#[must_use]
+pub fn rerun_if_git_changes() -> Option<()> {
+    // Make sure we get rerun when the git commit changes.
+    // First, find the `HEAD` file. This should work even with worktrees.
+    let output = Command::new("git").args(["rev-parse", "--git-dir"]).output().ok()?;
+    let stdout = output.status.success().then_some(output.stdout)?;
```

Comment:
Can it be a bit optimized like
```
let stdout = Command::new("git")
    .args(["rev-parse", "--git-dir"])
    .output()
    .ok()?
    .stdout;
```

Reply:
In which sense is that optimized? It's the same overall operations, and takes more lines of code...

Reply:
I meant instead of  
```
let output = Command::new("git").args(["rev-parse", "--git-dir"]).output().ok()?;
let stdout = output.status.success().then_some(output.stdout)?;
```
try this one
`let stdout = Command::new("git").args(["rev-parse", "--git-dir"]).output().ok()?.stdout;`

Reply:
That doesn't have the same behavior. We want to avoid looking at stdout when `success()` returns false.